pkg/loop/median_service.go

```diff
@@ -34,15 +34,13 @@ func NewMedianService(lggr logger.Logger, grpcOpts GRPCOpts, cmd func() *exec.Cm
 	lggr = logger.Named(lggr, "MedianService")
 	var ms MedianService
 	broker := BrokerConfig{StopCh: stopCh, Logger: lggr, GRPCOpts: grpcOpts}
-	ms.init(PluginMedianName, &GRPCPluginMedian{BrokerConfig: broker}, newService, lggr, cmd, stopCh)
+	ms.Init(PluginMedianName, &GRPCPluginMedian{BrokerConfig: broker}, newService, lggr, cmd, stopCh)
 	return &ms
 }
 
 func (m *MedianService) NewReportingPlugin(config ocrtypes.ReportingPluginConfig) (ocrtypes.ReportingPlugin, ocrtypes.ReportingPluginInfo, error) {
-	ctx, cancel := utils.ContextFromChan(m.pluginService.stopCh)
-	defer cancel()
-	if err := m.wait(ctx); err != nil {
+	if err := m.Wait(context.Background()); err != nil {
```

Comment:
Why a background context? Yielding to shutdown is important.

Reply:
I see that `Wait` yields to `stopCh` now, but then can we drop the context?

Reply:
I noticed that `m.wait` was invariably being called after getting the context from the `stopCh`, so rather than exposing that, moved this into the `wait` function.

I think it can still be useful for callers to override the timeout themselves, which is why I left context as part of the API, but can remove it if you prefer.

Reply:
Yeah I think remove since it is just creating noise? If there are cases with upstream contexts to pass in, then a `waitCtx` variant might actually be cleaner in order to avoid loose `context.Background()`s.